db_stress_tool/db_stress_driver.cc

```diff
@@ -253,22 +251,21 @@ bool RunStressTestImpl(SharedState* shared) {
       FLAGS_continuous_verification_interval > 0 ||
       FLAGS_compressed_secondary_cache_size > 0 ||
       FLAGS_compressed_secondary_cache_ratio > 0.0 ||
-      FLAGS_remote_compaction_worker_threads > 0) {
+      remote_compaction_worker_thread_count > 0) {
     MutexLock l(shared->GetMutex());
     shared->SetShouldStopBgThread();
     while (!shared->BgThreadsFinished()) {
       shared->GetCondVar()->Wait();
     }
   }
 
-  // Kill remote compaction workers
+  assert(remote_compaction_worker_threads.size() ==
+         remote_compaction_worker_thread_count);
```

Comment:
Would you mind reminding me again why did we use .capacity() in the previous PR? Also what is the root-cause of the failure and how did we fix it?

Reply:
Thanks would really appreciate some more explanation in the summary!